### ISSUE TITLE
Add authentication to login fakesmtp page

### ISSUE THREAD

**Issue by ilkin** [gessnerfl/fake-smtp-server#60]
Please add login page to authenticate fakesmtp interface. Authentication could be optional and Username and password can canfigure in application.properties.

**Comment by gessnerfl** [comment]
Will consider it in a future release. Contribution would also be appreciated ;)

**Comment by grig0ry** [comment]
+1, this would be useful

**Comment by gessnerfl** [comment]
@ilkin @grig0ry 
Do have any specific authentication method in mind? LDAP, OAuth, SAML2, etc.

**Comment by smuda** [comment]
Please make this optional. Our use case would be crippled with forced authentication.

**Comment by gessnerfl** [comment]
@smuda yes in case this is getting implemented authentication will be optional.

@ilkin / @grig0ry  can you please elaborate in more detail the need for authentication? The tool is intended for development purposes only.

**Comment by grig0ry** [comment]
@gessnerfl In our case, we run test builds of the application publicly, for QA and the development team. With a secure web UI, it would be calmer :)

**Comment by ilkin** [comment]
Also, we are using for QA and UAT purposes and it is public server or k8s container in company and many users can access, so simple authentication might be good at least.

**Comment by Slartibartfast27** [comment]
Just a thought: We also use fakesmtp, which contains non-public data in testcases. Therefore we added traefik as a reverse proxy with basic authentication before and made fakesmtp website accessible only through traefik, see https://doc.traefik.io/traefik/middlewares/http/basicauth/
Maybe this could be a simple solution without the need to change a line in fakesmtp.

**Comment by jessec** [comment]
Not sure if it is useful for anybody but you can implement a quick and dirty, but easy solution with this class.


package proxy;

import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpServer;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.InetSocketAddress;
import java.net.URL;
import java.util.List;
import java.util.Map;

public class ReverseProxyServer {

    private static final int PORT = 8000;
    private static final String DESTINATION_URL = "http://localhost:8080";

    public static void main(String[] args) throws IOException {
        HttpServer server = HttpServer.create(new InetSocketAddress(PORT), 0);
        server.createContext("/", new RequestHandler());
        server.start();
        System.out.println("Reverse Proxy Server started on port " + PORT);
    }

    static class RequestHandler implements HttpHandler {
        @Override
        public void handle(HttpExchange exchange) throws IOException {
            String requestMethod = exchange.getRequestMethod();
            String requestPath = exchange.getRequestURI().getPath();
            
            String requestQuery = exchange.getRequestURI().getQuery();
            
            
            System.out.println(requestQuery);
            
            String destinationUrl = DESTINATION_URL + requestPath + "?" + requestQuery;

            HttpURLConnection connection = (HttpURLConnection) new URL(destinationUrl).openConnection();
            connection.setRequestMethod(requestMethod);

            // Forward headers from the original request to the destination
            for (String header : exchange.getRequestHeaders().keySet()) {
                connection.setRequestProperty(header, exchange.getRequestHeaders().getFirst(header));
            }

            // Copy the cookies from the original request to the destination
            Map<String, List<String>> cookieHeaders = exchange.getRequestHeaders();
            if (cookieHeaders.containsKey("Cookie")) {
                String cookies = String.join("; ", cookieHeaders.get("Cookie"));
                System.out.println(cookies);
                connection.setRequestProperty("Cookie", cookies);
            }
            
            // Copy the response from the destination to the client
            try (OutputStream outputStream = exchange.getResponseBody();
                 InputStream inputStream = connection.getInputStream()) {
                exchange.sendResponseHeaders(connection.getResponseCode(), 0);
                byte[] buffer = new byte[4096];
                int bytesRead;
                while ((bytesRead = inputStream.read(buffer)) != -1) {
                    outputStream.write(buffer, 0, bytesRead);
                }
            } finally {
                connection.disconnect();
            }
        }
    }
}

**Comment by marcelgross90** [comment]
I added a PR for this: https://github.com/gessnerfl/fake-smtp-server/pull/749
maybe this could help ;)
